core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -318,6 +334,101 @@ private void rename0(Path path, CharSequence tableName, Path otherPath, CharSequ
         }
     }
 
+    public final TelemetryWriterJob startTelemetry() {
+        try {
+            this.telemetryWriterJob = new TelemetryWriterJob(configuration);
+        } catch (SqlException e) {
+            LOG.error().$("starting telemetry failed [error=").$(e.getMessage()).$(']').$();
+            throw CairoException.instance(0).put("Starting telemetry failed");
+        }
+
+        return this.telemetryWriterJob;
+    }
+
+    public final void storeTelemetry(short event) {
+        final MicrosecondClock clock = configuration.getMicrosecondClock();
+        final long cursor = telemetryPubSeq.next();
+        TelemetryRow row = telemetryQueue.get(cursor);
+
+        row.ts = clock.getTicks();
+        row.event = event;
+        telemetryPubSeq.done(cursor);
+    }
+
+    private class TelemetryWriterJob extends SynchronizedJob implements Closeable {
+        private final CharSequence telemetryTableName = "telemetry";
+        private final CharSequence telemetryConfigTableName = "telemetry_config";
+        private final QueueConsumer<TelemetryRow> myConsumer = this::toTelemetryTable;
+        private final StringSink idSink = new StringSink();
+        private final boolean telemetryEnabled = configuration.getTelemetryEnabled();
+        private final TableWriter writer;
+        private final TableWriter writerConfig;
+
+        public TelemetryWriterJob(CairoConfiguration configuration) throws SqlException {
+            sqlExecutionContext.with(AllowAllCairoSecurityContext.INSTANCE, null, null);
+
+            try(Path path = new Path()) {
+                final NanosecondClock nanosecondClock = configuration.getNanosecondClock();
+                final MicrosecondClock microsecondClock = configuration.getMicrosecondClock();
+
+                if (getStatus(AllowAllCairoSecurityContext.INSTANCE, path, telemetryConfigTableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                    final Long256Impl id = new Long256Impl();
+                    id.setLong0(nanosecondClock.getTicks());
```

Comment:
there is a way to set long256 on table writer once it is opened. This would make creating Long256Impl unnecessary.

Reply:
Done, it makes the code somehow convoluted, let me know what you think